### DISCUSSION TITLE
Reactivity : is Vue reactivity system faster than Signals ?

### DISCUSSION THREAD

**Discussion by edimitchel** [alpinejs/alpine#3981] [4. General]
I'm wondering if the choice of Alpine.js to use Vue's reactivity system is still the faster one comparing the Signal based system ?

It could be a good idea to compare those different system with an identical Alpine code.

**Comment by ekwoka** [comment]
No.

It isn't.

I have done the checks, and even implemented a signals based alternative that is faster and more memory safe than Vue reactivity for Alpine, but it's not tons faster.

Directly using, for example, Solid's amazing signals, wouldn't really be possible, or reasonably simple, due to some limitations Alpine assumes and requires that vue reactivity provides.

I'm aware that Vue is also using signals for a lot, and I'm not sure if they have a newer signals implementation that can satisfy those limitations, or maybe adaptations can be made to Solid's to satisfy those technical requirements.

Broadly, the reactivity engine is not the reason some things in Alpine are slower. There's a lot of other things that goes into Alpine that are outside of what other systems need to deal with (due to Alpine's role) that are more often the bottlenecks.

https://github.com/alpinejs/alpine/pull/3722

This PR had a significant impact on how expressions are processed.

Optimizations are a pretty tricky task that take a lot of knowledge (and a lot of measurements).

But broadly, if/when time is available to optimize Alpine, the reactivity engine is not going to be the most efficient use of that time. The difference is fairly small, vue reactivity is within the realm of similar implementations, and it satisfies the needs of the current system in a way the better ones do not clearly enable.

**Reply by edimitchel** [reply]
Got it, thank you for you clear explanation!

**Comment by horizoncarlo** [comment]
With Vapor Mode becoming available via [opt-in in Vue 3.6-alpha1](https://github.com/vuejs/core/releases/tag/v3.6.0-alpha.1) does anyone know how this will affect Alpine? Will it integrate the changes and core reactivity update that smooshed alien-signals into Vue?

**Reply by ekwoka** [reply]
Alpine is using a quite old version of Vue reactivity.
